### PR TITLE
Missing and redundant variable declarations

### PR DESCRIPTION
**Pull request by edwingustafson**
Add or remove variable declarations for variables ESLint found to be undeclared or redeclared

### CODE REVIEW

**Comment by scottleibrand** [comment]
Looks like checks are failing. Want to take a first look at that before I do?

File: lib/determine-basal/determine-basal.js [hunk 1]
```diff
@@ -566,22 +566,22 @@ var determine_basal = function determine_basal(glucose_status, currenttemp, iob_
...
             //console.error(predUCIslope, predUCImax);
             // predicted CI from UAM is the lesser of CI based on deviationSlope or DIA
-            predUCI = Math.min(predUCIslope, predUCImax);
+            var predUCI = Math.min(predUCIslope, predUCImax);
             if(predUCI>0) {
                 //console.error(UAMpredBGs.length,slopeFromDeviations, predUCI);
                 UAMduration=round((UAMpredBGs.length+1)*5/60,1);
             }
-            UAMpredBG = UAMpredBGs[UAMpredBGs.length-1] + predBGI + Math.min(0, predDev) + predUCI;
+            var UAMpredBG = UAMpredBGs[UAMpredBGs.length-1] + predBGI + Math.min(0, predDev) + predUCI;
```
Comment (cluckj) [inline lib/determine-basal/determine-basal.js]: I think this `var` is messing up determine-basal, as I'm getting the following error:
```
/usr/local/lib/node_modules/oref0/lib/determine-basal/determine-basal.js:703
        avgPredBG = round( (IOBpredBG + UAMpredBG)/2 );
                                        ^

ReferenceError: UAMpredBG is not defined
    at determine_basal (/usr/local/lib/node_modules/oref0/lib/determine-basal/determine-basal.js:703:41)
    at Object.<anonymous> (/usr/local/lib/node_modules/oref0/bin/oref0-determine-basal.js:215:14)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
    at bootstrap_node.js:609:3
No deliverAt found.
Couldn't smb_verify_suggested
```

File: lib/determine-basal/determine-basal.js [hunk 2]
```diff
@@ -566,22 +566,22 @@ var determine_basal = function determine_basal(glucose_status, currenttemp, iob_
             remainingCIs.push(round(remainingCI,0));
             predCIs.push(round(predCI,0));
             //process.stderr.write(round(predCI,1)+"+"+round(remainingCI,1)+" ");
-            COBpredBG = COBpredBGs[COBpredBGs.length-1] + predBGI + Math.min(0,predDev) + predCI + remainingCI;
-            aCOBpredBG = aCOBpredBGs[aCOBpredBGs.length-1] + predBGI + Math.min(0,predDev) + predACI;
+            var COBpredBG = COBpredBGs[COBpredBGs.length-1] + predBGI + Math.min(0,predDev) + predCI + remainingCI;
```
Comment (cluckj) [inline lib/determine-basal/determine-basal.js]: Related to comment below, determine-basal crashes with this declaration:
```
/usr/local/lib/node_modules/oref0/lib/determine-basal/determine-basal.js:697
        avgPredBG = round( (1-fractionCarbsLeft)*UAMpredBG + fractionCarbsLeft*COBpredBG );
                                                                               ^

ReferenceError: COBpredBG is not defined
    at determine_basal (/usr/local/lib/node_modules/oref0/lib/determine-basal/determine-basal.js:697:80)
    at Object.<anonymous> (/usr/local/lib/node_modules/oref0/bin/oref0-determine-basal.js:215:14)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
    at bootstrap_node.js:609:3
No deliverAt found.
```
Removing it fixes the issue.

**Comment by edwingustafson** [comment]
Not sure why neither ESLint nor the project's unit tests reported these in my fork, but it looks like your pull requests 1198 and 1199 solved them. Sorry for the trouble!

**Pull request merged**